### ISSUE TITLE
When first cluster node dies, transaction is commited.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sagarswathi/h2database#199]
```

What steps will reproduce the problem?
(simple SQL scripts or simple standalone applications are preferred)
1. initialize cluster.
2. execute insert or update query.
3. kill the first node cluster. transaction is commited without commit
statement.

What is the expected output? What do you see instead?
can rollback or commit transaction.

What version of the product are you using? On what operating system, file
system, and virtual machine?

h2 1.2.135
Windows XP SP3
java version "1.6.0_18"

Do you know a workaround?

How important/urgent is the problem for you?

In your view, is this a defect or a feature request?

defect

Please provide any additional information below.

```

Original issue reported on code.google.com by `litail...@gmail.com` on 17 May 2010 at 12:28

**Comment by GoogleCodeExporter** [comment]
```
Did you disable auto-commit? If yes, could you you post a complete reproducible 
test
case please (SQL script and/or a Java class with a main method and without 
dependencies)?

```

Original comment by `thomas.t...@gmail.com` on 20 May 2010 at 4:51

**Comment by GoogleCodeExporter** [comment]
```
Hi.

> Did you disable auto-commit? 

Yes.


create cluster.

java -cp d:\temp\h2-1.2.135.jar org.h2.tools.Server -tcp -tcpPort 9101 -baseDir 
D:\temp\server1
java -cp d:\temp\h2-1.2.135.jar org.h2.tools.Server -tcp -tcpPort 9102 -baseDir 
D:\temp\server2
java -cp d:\temp\h2-1.2.135.jar org.h2.tools.CreateCluster -urlSource 
jdbc:h2:tcp://localhost:9101/testdb -urlTarget 
jdbc:h2:tcp://localhost:9102/testdb -
user admin -password admin -serverList localhost:9101,localhost:9102

connect to clustered database.

-------------
D:\temp>java -cp d:\temp\h2-1.2.135.jar org.h2.tools.Shell -url 
jdbc:h2:tcp://localhost:9101,localhost:9102/testdb -user admin -password admin

Welcome to H2 Shell 1.2.135 (2010-05-08)
Exit with Ctrl+C
Commands are case insensitive; SQL statements end with ';'
help or ?      Display this help
list           Toggle result list / stack trace mode
maxwidth       Set maximum column width (default is 100)
show           List all tables
describe       Describe a table
history        Show the last 20 statements
quit or exit   Close the connection and exit

sql> set autocommit false;
(Update count: 0, 0 ms)
sql> create table test(id int);
(Update count: 0, 0 ms)
sql> insert into test select x from system_range(1,2000);
(Update count: 2000, 203 ms)
sql> select count(*) from test;
COUNT(*)
2000
(1 row, 15 ms)
sql> rollback;
(Update count: 0, 31 ms)
sql> select count(*) from test;
COUNT(*)
0
(1 row, 0 ms)
sql> insert into test select x from system_range(1,2000);
(Update count: 2000, 47 ms)

<<<<< then I killed server1. >>>>

sql> select count(*) from test;
COUNT(*)
2000
(1 row, 16 ms)
sql> rollback;
(Update count: 0, 0 ms)
sql> select count(*) from test;
COUNT(*)
2000
(1 row, 0 ms)
sql>

---------------


```

Original comment by `litail...@gmail.com` on 20 May 2010 at 6:34

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Thanks a lot! I think the problem is that disabling the cluster (after killing 
the
second server) commits the transaction. I think I know how to fix the problem 
now.

Regards,
Thomas

```

Original comment by `thomas.t...@gmail.com` on 28 May 2010 at 2:16
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This should be fixed now in version 1.2.137
```

Original comment by `thomas.t...@gmail.com` on 6 Jun 2010 at 10:49
- Changed state: **Fixed**
